DLP-2 Invitation, DLP-3 Program, DLP-4 Login

KristofLevente opened the pull request.

KristofLevente committed:
DLP-2 Invitation, DLP-3 Program, DLP-4 Login

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,69 @@
+        }
+
+        $program = $request->user()->program()->create([
+            'opening_hour' => $request['opening_hour'],
```
Please use `$request->input('opening_hour')` instead of this.

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,69 @@
+            $request->user()->invites()->createMany($invitations);
+
+        foreach ($invitations as $invitation) {
+            Notification::route('mail', $invitation['email'])->notify(new InvitationNotification($invitation['token']));
```
If you specify the route explicitly in the `InvitationNotification` class, then you don't need to add the `Notification::route('mail'`. You can use the `Notification::send(` function

https://laravel.com/docs/11.x/notifications#using-the-notification-facade

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,69 @@
+
+            $request->user()->invites()->createMany($invitations);
+
+        foreach ($invitations as $invitation) {
```
You can still use the previously created collection here. If you have a collection and you want to iterate through you can use the `map` method instead of the regular foreach

arnold-webgurus reviewed (changes requested):
Please check my comments.

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,70 @@
+                    'role' => $invitation['role'],
+                    'token' => Str::random(50),
+                ];
+            })->all();
```
If you delete the `all` method call you still have a collection. And you can use it for later.

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,70 @@
+
+            $inv = $request->user()->invites()->createMany($invitations);
+
+            collect($inv)->map(function ($invitation){
```
You can use the previously created `$invitations` collection for the mapping.

arnold-webgurus commented on app/Http/Controllers/ClinicProgramController.php:
```diff
@@ -0,0 +1,70 @@
+                ];
+            })->all();
+
+            $inv = $request->user()->invites()->createMany($invitations);
```
This is redundant.

KristofLevente committed:
Fixing comments

- deleted the all() function after the invitation collection
-made variable names more descriptive
-removed the map() function from the Notification and gave all the objects to the send() function instead
